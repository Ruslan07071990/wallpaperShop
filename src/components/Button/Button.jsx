import React from "react";
import styles from "./Button.module.scss";
import PropTypes from "prop-types";
// import { type } from "@testing-library/user-event/dist/type";
const Button = (props) => {
  const { backgroundColor, text, onClick,type,disabled } = props;

  return (
    <button
      className={styles.btn}
      style={{ backgroundColor }}
      onClick={onClick}
      type={type}
      disabled={disabled}
    >
      {text}
     
    </button>
  );
};
Button.propTypes = {
  backgroundColor: PropTypes.string,
  text: PropTypes.string,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  type: "button",
  onClick: () => {},
};

export default Button;
