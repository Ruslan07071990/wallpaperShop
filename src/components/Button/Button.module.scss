.btn {
    position: relative;
    // width: 170px;
    margin: 15px;
    padding: 10px;
    
    text-align: center; 
    box-sizing: border-box;
    align-items: center; 
    justify-content: center;
    border: none;
    border-radius: 4px;
   
    font-family: Roboto,sans-serif;
    font-weight: 400;
    font-size: 24px;
   
    cursor: pointer;
   
      
    
  }
 .btn:hover{
 background-color: chartreuse;}
  
  .btn-primary {
    color: blue;
    text-transform: uppercase;
    font-weight: 600;
  }
  
  .btn-secondary {
    color: yellow;
  }
  
  