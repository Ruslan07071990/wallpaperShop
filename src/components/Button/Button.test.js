import { render } from "@testing-library/react";
import Button from "./Button";


describe("Snapshot test", () => {
    test("should Item render", () => {
      const { asFragment } = render(<Button type="button" text="Hello"> Click </Button>);
      expect(asFragment()).toMatchSnapshot();
    });
  });