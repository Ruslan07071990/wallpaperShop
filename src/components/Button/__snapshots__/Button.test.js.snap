// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test should Item render 1`] = `
<DocumentFragment>
  <button
    class="btn"
    type="button"
  >
    Hello
  </button>
</DocumentFragment>
`;
